### ISSUE TITLE
Cannot reproduce the result on AVA

### ISSUE THREAD

**Issue by taosean** [facebookresearch/SlowFast#141]
Hi, thank you for your great code base.

Now I'm trying to reproduce the result on AVA claimed in paper and tech report. However, I cannot reproduce the result.
I chose [configs/AVA/c2/SLOWFAST_32x2_R101_50_50.yaml](https://github.com/facebookresearch/SlowFast/blob/master/configs/AVA/c2/SLOWFAST_32x2_R101_50_50.yaml) configuration file for training.
The modifications I did to this config file are: (**Here, the commented are original settings**.)
```yaml
TRAIN:
  ENABLE: True #False
  DATASET: ava
  BATCH_SIZE: 80 #16
  EVAL_PERIOD: 2 # 1
  CHECKPOINT_PERIOD: 1
  AUTO_RESUME: True
  CHECKPOINT_FILE_PATH: path to pretrain model downloaded from the third entry in this table(https://github.com/facebookresearch/SlowFast/blob/master/MODEL_ZOO.md#ava)
  CHECKPOINT_TYPE: caffe2 # pytorch
DETECTION:
  ENABLE: True
  ALIGNED: True #False
#SOLVER:
#  MOMENTUM: 0.9
#  WEIGHT_DECAY: 1e-7
#  OPTIMIZING_METHOD: sgd
SOLVER:
  BASE_LR: 0.1
  LR_POLICY: steps_with_relative_lrs
  STEPS: [0, 20, 30]
  LRS: [1, 0.1, 0.01, 0.001]
  MAX_EPOCH: 40
  MOMENTUM: 0.9
  WEIGHT_DECAY: 1e-7
  WARMUP_EPOCHS: 5
  WARMUP_START_LR: 0.000125
  OPTIMIZING_METHOD: sgd
```

As you can see, I modified the learning policy from `cosine` to `steps_with_relative_lrs`, which is stated in paper, with reference to the settings in https://github.com/facebookresearch/SlowFast/blob/a8a47ced376a681e76d8b904e7be76d67fe999b3/configs/AVA/SLOWFAST_32x2_R50_SHORT.yaml#L52-L62
However, the result is not good.

Below is the picture of training loss and mAP on val set, 
![批注 2020-03-10 134205](https://user-images.githubusercontent.com/17960909/76285609-21a45880-62db-11ea-9557-7aa169a48ffe.jpg)

From the picture, we can find that the mAP on Val set saturates around `26 .0` quickly and cannot reach the value (**`29.0`**) stated in paper.

How do you think about this, is this because of the `STEPS` parameter I set that cause overfitting?

Thank you!

**Comment by takatosp1** [comment]
Hi 
Thanks for playing with PySlowFast!
I noticed you've change the batch size, as well as the ALIGN, they all affect the performance significantly. Also I am curious that how could you run a SlowFast R101 model with batch size of 80? Could you share me what are the number of GPUs you use?

btw, note that, for the scripts under `/c2/`, they are script designed to use for inference.

**Comment by taosean** [comment]
Hi @takatosp1 , thank you for your reply.

I'm training my model on a machine with 8 Quadro RTX 6000 GPUs (24G memory).

---
> I noticed you've change the batch size, as well as the ALIGN, they all affect the performance significantly.

I noticed that in 
https://github.com/facebookresearch/SlowFast/blob/a8a47ced376a681e76d8b904e7be76d67fe999b3/configs/AVA/SLOWFAST_32x2_R50_SHORT.yaml#L1-L9

that batchsize is set to `64`, since my batchsize is `80`, now I'm doing another training with 
```yaml
DETECTION:
  ENABLE: True
  ALIGNED: True
SOLVER:
  BASE_LR: 0.125 #0.1
  LR_POLICY: steps_with_relative_lrs
  STEPS: [0, 8, 12]
  LRS: [1, 0.1, 0.01, 0.001]
  MAX_EPOCH: 16
  MOMENTUM: 0.9
  WEIGHT_DECAY: 1e-7
  WARMUP_EPOCHS: 5
  WARMUP_START_LR: 0.000125
  OPTIMIZING_METHOD: sgd
```

with the
```yaml
  BASE_LR: 0.125 #0.1
  STEPS: [0, 8, 12]
  LRS: [1, 0.1, 0.01, 0.001]
  MAX_EPOCH: 16
```
calculated according to 
https://github.com/facebookresearch/SlowFast/blob/a8a47ced376a681e76d8b904e7be76d67fe999b3/configs/AVA/SLOWFAST_32x2_R50_SHORT.yaml#L52-L62
are these settings correct?

Besides, how is `ALIGN` going to affect the performance? Are the models claimed in paper trained with `ALIGN` set to `True` or `False`?

---

>btw, note that, for the scripts under /c2/, they are script designed to use for inference.

Thank you for reminding. 
However, if so, there are no scripts for training with R101 model, right? It it OK if I modify the inference script for training?

Best regards!

**Comment by tonysy** [comment]
@takatosp1  I noticed you use ALIGNED = False for all provided models in /c2, while for the c2d_short and slowfast_shot , you use ALIGNED = True, so which option is correct for training a pytorch model with you codebase? thanks a lot

**Comment by takatosp1** [comment]
Hi @tonysy 
According to your email, for your case I don't think the 10% off is caused by the ALIGNED. 
ALIGNED would not case a big different if you try to train it by your self. 
Feel free to raise your further questions in **github**.

**Comment by leviswind** [comment]
@taosean  The link of the slowfast-101 pretrained model on kinetics dataset is not valid. Do you train your model on AVA dataset from scratch?

**Comment by takatosp1** [comment]
We will release the R101 models later. (The link says coming soon) 📦 

We were discussion the performance on SlowFast50 models.

Thanks

**Comment by leviswind** [comment]
@takatosp1 Thanks. Also, can you explain the meaning of * in here? What's the difference of used training region proposals?
![image](https://user-images.githubusercontent.com/11644170/77984266-a404c300-7343-11ea-87ae-fd741c8e9cf7.png)

**Comment by chaoyuaw** [comment]
> We will release the R101 models later. (The link says coming soon) 📦
> 
> We were discussion the performance on SlowFast50 models.
> 
> Thanks

I'm a bit confused. Why do we depend on the "coming soon" models? To reproduce the AVA mAP (e.g., the "29.1" and "29.4" in this table), shouldn't we just use the pre-trained models provided by the "Pretrain Model" column (red box on the attached image) in the table? 
![Screen Shot 2020-03-31 at 10 47 40 AM](https://user-images.githubusercontent.com/1841547/78047435-f3e08b80-733d-11ea-8425-ffcc19047d89.png)

**Comment by chaoyuaw** [comment]
> Hi @tonysy
> According to your email, for your case I don't think the 10% off is caused by the ALIGNED.
> ALIGNED would not case a big different if you try to train it by your self.
> Feel free to raise your further questions in **github**.

I also see the same performance drop when trying to reproduce the "29.1" model. I will describe my exact settings below. I would really appreciate if anyone can provide any hints. Thanks in advance. 

I first take AVA/c2/SLOWFAST_32x2_R101_50_50.yaml and add the training-related options to it from AVA/SLOWFAST_32x2_R50_SHORT.yaml, since the c2/R101 model only support inference. 

I verify that I can obtain correct mAP (29.1) by running testing on AVA/c2/SLOWFAST_32x2_R101_50_50.yaml using the trained weights provided MODEL-ZOO, so my data should be correct.

I also verified that I was able to train AVA/SLOWFAST_32x2_R50_SHORT.yaml and obtain 24.9 (with test crop size 256), so the training code also works at my end. The only thing I wasn't sure about is the schedule. In the paper 68 epochs were used. 
![Screen Shot 2020-03-31 at 11 03 49 AM](https://user-images.githubusercontent.com/1841547/78048440-5a19de00-733f-11ea-9019-c213eeaf600a.png)
but I see the same thing as @taosean that it saturates in around 10-15 epochs. With my tuned schedule (around 25-30 epochs in total) I obtain at most 27.0 mAP.

**Comment by chaoyuaw** [comment]
@taosean, since there're multiple follow-up questions related to this issue. Shall we reopen this issue?

**Comment by tonysy** [comment]
@chaoyuaw Thanks for your reply, I have resolved the performance issue on my side. (#112)

**Comment by chaoyuaw** [comment]
> @chaoyuaw Thanks for your reply, I have resolved the performance issue on my side. (#112)

@tonysy , Thanks for sharing! Glad that you resolved the issue!  It seems that the results I see are specific to the larger SLOWFAST_32x2_R101_50_50 model. I was wondering if you've ever trained an R101-K600-pretrained model. Thanks! :)

**Comment by tonysy** [comment]
@chaoyuaw Training R101 model is time-consuming, I will report the performance of  `SLOWFAST_32x2_R101_50_50` once I finish the training. I have verified the configuration `configs/AVA/SLOWFAST_32x2_R50_SHORT.yaml` and reported the performance in **https://github.com/facebookresearch/SlowFast/issues/112**.

If some guys meet the similar problem, the training log and test log are listed here for reference.
| architecture | depth | Pretrain Model |  frame length x sample rate  | MAP | AVA version | Log | config |
| ------------- | ------------- | ------------- | ------------- | ------------- | ------------- |------------- |------------- |
| C2D      | R50 | [C2D(8x8)-Kinetics 400](https://dl.fbaipublicfiles.com/pyslowfast/model_zoo/ava/pretrain/C2D_8x8_R50.pkl) | 4 x 16 |  20.69    | 2.2 | [[train log]](https://gist.githubusercontent.com/tonysy/6c8123089a7b069c7c9236e8550dbc46/raw/31a6b12b5351f906e71098216571ed1f5fc0fe71/c2d.ava.8x8.res50.short_train_log.txt) [[test log]](https://gist.githubusercontent.com/tonysy/bbf2df67b10ebe6bc7a1352d9cc7f269/raw/6394b111f8c447e6b3bace7f1b73ac2b927f5352/c2d.ava.8x8.res50.short_test_log.txt)| workspace/ava/c2d.ava.8x8.res50.short
| SlowFast      | R50 | [SlowFast(8x8)-Kinetics 400](https://dl.fbaipublicfiles.com/pyslowfast/model_zoo/kinetics400/SLOWFAST_8x8_R50.pkl) | S:8 x 8;F:32x2 |  24.9    | 2.2 | [[train log]](https://gist.githubusercontent.com/tonysy/bedf4f876f6b6646efc51f9f761682be/raw/19129f535ab61e38d7e3b8c48490bba7c6e73e64/slowfast.ava.32x2.res50.short_train_log.txt)[[test log]](https://gist.githubusercontent.com/tonysy/ee5c8eaeaf16d1a1acb313376953b8cd/raw/259db16f7c699c306497eb48f1959b000d11f17e/slowfast.ava.32x2.res50.short_test_log.txt)| workspace/ava/slowfast.ava.32x2.res50.short

**Comment by takatosp1** [comment]
Thanks for sharing it!

**Comment by taosean** [comment]
> The link of the slowfast-101 pretrained model on kinetics dataset is not valid. Do you train your model on AVA dataset from scratch?
The performance I posted in this issue is not trained from scratch, it uses the pretrained model provided in MODEL ZOO.

**Comment by taosean** [comment]
> @taosean, since there're multiple follow-up questions related to this issue. Shall we reopen this issue?

This issue is not  closed by me. so it seems I cannot reopen this issue.

**Comment by takatosp1** [comment]
Hi

We haven't release R101 training schedule and will do it later.

**Comment by BoPang1996** [comment]
Hi, @chaoyuaw. I have met the same issue.  Only SlowFast-R101 8x8 cannot be reproduced and I only got 27.2 mAP. Have you solved this problem?


> > Hi @tonysy
> > According to your email, for your case I don't think the 10% off is caused by the ALIGNED.
> > ALIGNED would not case a big different if you try to train it by your self.
> > Feel free to raise your further questions in **github**.
> 
> I also see the same performance drop when trying to reproduce the "29.1" model. I will describe my exact settings below. I would really appreciate if anyone can provide any hints. Thanks in advance.
> 
> I first take AVA/c2/SLOWFAST_32x2_R101_50_50.yaml and add the training-related options to it from AVA/SLOWFAST_32x2_R50_SHORT.yaml, since the c2/R101 model only supports inference.
> 
> I verify that I can obtain correct mAP (29.1) by running testing on AVA/c2/SLOWFAST_32x2_R101_50_50.yaml using the trained weights provided in MODEL-ZOO, so my data should be correct.
> 
> I also verified that I was able to train AVA/SLOWFAST_32x2_R50_SHORT.yaml and obtain 24.9 (with test crop size 256), so the training code also works at my end. The only thing I wasn't sure about is the schedule. In the paper 68 epochs were used.
> ![Screen Shot 2020-03-31 at 11 03 49 AM](https://user-images.githubusercontent.com/1841547/78048440-5a19de00-733f-11ea-9019-c213eeaf600a.png)
> but I see the same thing as @taosean that it saturates in around 10-15 epochs. With my tuned schedule (around 25-30 epochs in total) I obtain at most 27.0 mAP.

**Comment by BoPang1996** [comment]
Hi, @takatosp1 @feichtenhofer.

I notice that in the paper of SlowFast, SlowFast-R101, 8x8, K600 achieves 29.0 on AVA-v2.2, and in the paper of X3D, the performance is reported as 27.4 for SlowFast-R101, 8x8, K600. What is the difference between their training and inference settings?

**Comment by ecr23xx** [comment]
> Hi, @takatosp1 @feichtenhofer.
> 
> I notice that in the paper of SlowFast, SlowFast-R101, 8x8, K600 achieves 29.0 on AVA-v2.2, and in the paper of X3D, the performance is reported as 27.4 for SlowFast-R101, 8x8, K600. What is the difference between their training and inference settings?

Same question here.

**Comment by ShoufaChen** [comment]
> @chaoyuaw Training R101 model is time-consuming, I will report the performance of `SLOWFAST_32x2_R101_50_50` once I finish the training. I have verified the configuration `configs/AVA/SLOWFAST_32x2_R50_SHORT.yaml` and reported the performance in **#112**.
> 
> If some guys meet the similar problem, the training log and test log are listed here for reference.(I will update the performance for R101 later)
> 
> architecture	depth	Pretrain Model	frame length x sample rate	MAP	AVA version	Log	config
> C2D	R50	[C2D(8x8)-Kinetics 400](https://dl.fbaipublicfiles.com/pyslowfast/model_zoo/ava/pretrain/C2D_8x8_R50.pkl)	4 x 16	20.69	2.2	[[train log]](https://gist.githubusercontent.com/tonysy/6c8123089a7b069c7c9236e8550dbc46/raw/31a6b12b5351f906e71098216571ed1f5fc0fe71/c2d.ava.8x8.res50.short_train_log.txt) [[test log]](https://gist.githubusercontent.com/tonysy/bbf2df67b10ebe6bc7a1352d9cc7f269/raw/6394b111f8c447e6b3bace7f1b73ac2b927f5352/c2d.ava.8x8.res50.short_test_log.txt)	workspace/ava/c2d.ava.8x8.res50.short
> SlowFast	R50	[SlowFast(8x8)-Kinetics 400](https://dl.fbaipublicfiles.com/pyslowfast/model_zoo/kinetics400/SLOWFAST_8x8_R50.pkl)	S:8 x 8;F:32x2	24.9	2.2	[[train log]](https://gist.githubusercontent.com/tonysy/bedf4f876f6b6646efc51f9f761682be/raw/19129f535ab61e38d7e3b8c48490bba7c6e73e64/slowfast.ava.32x2.res50.short_train_log.txt)[[test log]](https://gist.githubusercontent.com/tonysy/ee5c8eaeaf16d1a1acb313376953b8cd/raw/259db16f7c699c306497eb48f1959b000d11f17e/slowfast.ava.32x2.res50.short_test_log.txt)	workspace/ava/slowfast.ava.32x2.res50.short

@tonysy Hi, from the test log you provided, I noticed the config `TRAIN.ENABLE` is `True`, but the log only contains the inference information. I wonder what's the reason.  Thank you in advance.

**Comment by gurkirt** [comment]
> Hi
> 
> We haven't release R101 training schedule and will do it later.
> Please not use Recipe under c2 for training since it is for inference only

Any update here?

**Comment by YuJiZhe** [comment]
> Hi, thank you for your great code base.
> 
> Now I'm trying to reproduce the result on AVA claimed in paper and tech report. However, I cannot reproduce the result.
> I chose [configs/AVA/c2/SLOWFAST_32x2_R101_50_50.yaml](https://github.com/facebookresearch/SlowFast/blob/master/configs/AVA/c2/SLOWFAST_32x2_R101_50_50.yaml) configuration file for training.
> The modifications I did to this config file are: (**Here, the commented are original settings**.)
> 
> ```yaml
> TRAIN:
>   ENABLE: True #False
>   DATASET: ava
>   BATCH_SIZE: 80 #16
>   EVAL_PERIOD: 2 # 1
>   CHECKPOINT_PERIOD: 1
>   AUTO_RESUME: True
>   CHECKPOINT_FILE_PATH: path to pretrain model downloaded from the third entry in this table(https://github.com/facebookresearch/SlowFast/blob/master/MODEL_ZOO.md#ava)
>   CHECKPOINT_TYPE: caffe2 # pytorch
> DETECTION:
>   ENABLE: True
>   ALIGNED: True #False
> #SOLVER:
> #  MOMENTUM: 0.9
> #  WEIGHT_DECAY: 1e-7
> #  OPTIMIZING_METHOD: sgd
> SOLVER:
>   BASE_LR: 0.1
>   LR_POLICY: steps_with_relative_lrs
>   STEPS: [0, 20, 30]
>   LRS: [1, 0.1, 0.01, 0.001]
>   MAX_EPOCH: 40
>   MOMENTUM: 0.9
>   WEIGHT_DECAY: 1e-7
>   WARMUP_EPOCHS: 5
>   WARMUP_START_LR: 0.000125
>   OPTIMIZING_METHOD: sgd
> ```
> 
> As you can see, I modified the learning policy from `cosine` to `steps_with_relative_lrs`, which is stated in paper, with reference to the settings in
> 
> https://github.com/facebookresearch/SlowFast/blob/a8a47ced376a681e76d8b904e7be76d67fe999b3/configs/AVA/SLOWFAST_32x2_R50_SHORT.yaml#L52-L62
> 
> 
> However, the result is not good.
> Below is the picture of training loss and mAP on val set,
> ![批注 2020-03-10 134205](https://user-images.githubusercontent.com/17960909/76285609-21a45880-62db-11ea-9557-7aa169a48ffe.jpg)
> 
> From the picture, we can find that the mAP on Val set saturates around `26 .0` quickly and cannot reach the value (**`29.0`**) stated in paper.
> 
> How do you think about this, is this because of the `STEPS` parameter I set that cause overfitting?
> 
> Thank you!

Hi. can u give me some advice. please, sorry for my poor English.[(https://github.com/facebookresearch/SlowFast/issues/469)]
